stepper-lib/src/air/fold/utils.rs

```diff
@@ -73,78 +63,121 @@ fn handle_instruction_variable<'ctx>(
-
             let acc = acc.borrow();
             if acc.is_empty() {
                 return Ok(None);
             }
 
             let acc_iter = acc.iter().map(|v| v.result.deref());
-            let (jvalues, tetraplet_indices) = select_with_iter(acc_iter, &path)
-                .map_err(|e| JValueAccJsonPathError(acc.clone(), path.to_string(), e))?;
+            let (jvalues, tetraplet_indices) = select_with_iter(acc_iter, &json_path)
+                .map_err(|e| JValueAccJsonPathError(acc.clone(), json_path.to_string(), e))?;
 
             let jvalues = jvalues.into_iter().cloned().collect();
             let tetraplets = tetraplet_indices
                 .into_iter()
                 .map(|id| SecurityTetraplet {
                     triplet: acc[id].triplet.clone(),
-                    json_path: path.to_string(),
+                    json_path: json_path.to_string(),
                 })
                 .collect::<Vec<_>>();
 
             let foldable = IterableVecJsonPathResult::init(jvalues, tetraplets);
-            Box::new(foldable)
+            Some(Box::new(foldable))
         }
-        _ => {
-            return Err(AquamarineError::InstructionError(String::from(
-                "At now, it isn't possible to use fold iterator in other folds",
-            )))
+        Some(AValue::JValueFoldCursor(fold_state)) => {
+            let iterable_value = fold_state.iterable.peek().unwrap();
+            let jvalues = iterable_value.apply_json_path(json_path)?;
+            let triplet = as_triplet(&iterable_value);
+
+            from_jvalues(jvalues, triplet, json_path)
         }
+        _ => return Err(AquamarineError::VariableNotFound(variable_name.to_string())),
+    };
+
+    Ok(iterable)
+}
+
+fn apply_json_path<'jvalue, 'str>(jvalue: &'jvalue JValue, json_path: &'str str) -> Result<Vec<&'jvalue JValue>> {
+    use AquamarineError::JValueJsonPathError;
+
+    select(jvalue, json_path).map_err(|e| JValueJsonPathError(jvalue.clone(), json_path.to_string(), e))
+}
+
+/// Applies json_path to provided jvalues and construct IterableValue from the result and given triplet.
+fn from_jvalues(jvalues: Vec<&JValue>, triplet: Rc<ResolvedTriplet>, json_path: &str) -> Option<IterableValue> {
+    if jvalues.is_empty() {
+        return None;
+    }
+
+    let jvalues = jvalues.into_iter().cloned().collect();
+
+    let tetraplet = SecurityTetraplet {
+        triplet,
+        json_path: json_path.to_string(),
     };
 
-    Ok(Some(iterable))
+    let foldable = IterableJsonPathResult::init(jvalues, tetraplet);
+    Some(Box::new(foldable))
+}
+
+fn as_triplet(iterable: &IterableItem<'_>) -> Rc<ResolvedTriplet> {
+    use IterableItem::*;
+
+    // clone is cheap here, because triplet is under Rc
+    match iterable {
+        RefRef((_, tetraplet)) => tetraplet.triplet.clone(),
+        RefValue((_, tetraplet)) => tetraplet.triplet.clone(),
+        RcValue((_, tetraplet)) => tetraplet.triplet.clone(),
+    }
```

Comment:
```suggestion
    let tetraplet = match iterable {
        RefRef((_, tetraplet)) => tetraplet,
        RefValue((_, tetraplet)) => tetraplet,
        RcValue((_, tetraplet)) => tetraplet,
    };

    // clone is cheap here, because triplet is under Rc
    Rc::clone(&tetraplet.triplet)
```